test/manifest.spec.js

```diff
@@ -101,3 +102,36 @@ test("manifest.readAnalyze should return hasNativeElements: true because of the
     readFile.restore();
   }
 });
+
+test("manifest.readAnalyze should generate the proper integrity", async() => {
+  // Warning: property must be in the same order
+  const manifestJSON = {
+    name: "foo",
+    version: "1.0.0",
+    dependencies: {
+      "node-addon-api": "^1.0.0"
+    },
+    license: "MIT",
+    scripts: {
+      test: "hello world!"
+    }
+  };
+  const expectedIntegrity = crypto
+    .createHash("sha256")
+    .update(JSON.stringify(manifestJSON))
+    .digest("hex");
```

Comment:
Maybe we should expose (not to the public API) a `generateIntegrityHash` method we could reuse here?

Reply:
Didn't abstract it cause only used it two times. Also input and context in both cases were not exactly the same.

And having a function just to abstract crypto.createHash does not feel useful to me.